test/shared/src/main/scala/zio/test/mock/MockRandom.scala

```diff
@@ -26,146 +30,173 @@ trait MockRandom extends Random {
+      } yield ((i1.toDouble * (1L << 27).toDouble) + i2.toDouble) / (1L << 53).toDouble
+
+    def nextBytes(length: Int): UIO[Chunk[Byte]] = {
+      //  Our RNG generates 32 bit integers so to maximize efficieny we want to
+      //  pull 8 bit bytes from the current integer until it is exhausted
+      //  before generating another random integer
+      def loop(i: Int, rnd: UIO[Int], n: Int, acc: UIO[List[Byte]]): UIO[List[Byte]] =
+        if (i == length)
+          acc.map(_.reverse)
+        else if (n > 0)
+          rnd.flatMap(rnd => loop(i + 1, UIO.succeed(rnd >> 8), n - 1, acc.map(rnd.toByte :: _)))
+        else
+          loop(i, nextInt, (length - i) min 4, acc)
+
+      loop(0, nextInt, length min 4, UIO.succeed(List.empty[Byte])).map(Chunk.fromIterable)
+    }
 
-    val clearStrings: UIO[Unit] =
-      randomState.update(data => data.copy(strings = List.empty)).unit
+    val nextFloat: UIO[Float] =
+      next(24).map(i => (i.toDouble / (1 << 24).toDouble).toFloat)
+
+    val nextGaussian: UIO[Double] =
+      //  The Box-Muller transform generates two normally distributed random
+      //  doubles, so we store the second double in a queue and check the
+      //  queue before computing a new pair of values to avoid wasted work.
+      randomState.modify {
+        case Data(seed1, seed2, queue) =>
+          queue.dequeueOption.fold { (Option.empty[Double], Data(seed1, seed2, queue)) } {
+            case (d, queue) => (Some(d), Data(seed1, seed2, queue))
+          }
+      }.flatMap {
+        case Some(nextNextGaussian) => UIO.succeed(nextNextGaussian)
+        case None =>
+          def loop: UIO[(Double, Double, Double)] =
+            nextDouble.zip(nextDouble).flatMap {
+              case (d1, d2) =>
+                val x      = 2 * d1 - 1
+                val y      = 2 * d2 - 1
+                val radius = x * x + y * y
+                if (radius >= 1 || radius == 0) loop else UIO.succeed((x, y, radius))
+            }
+          loop.flatMap {
+            case (x, y, radius) =>
+              val c = sqrt(-2 * log(radius) / radius)
+              randomState.modify {
+                case Data(seed1, seed2, queue) =>
+                  (x * c, Data(seed1, seed2, queue.enqueue(y * c)))
+              }
+          }
+      }
 
-    val clearBytes: UIO[Unit] =
-      randomState.update(data => data.copy(bytes = List.empty)).unit
+    val nextInt: UIO[Int] =
+      next(32)
+
+    def nextInt(n: Int): UIO[Int] =
+      if (n <= 0)
+        UIO.die(new IllegalArgumentException("n must be positive"))
+      else if ((n & -n) == n)
+        next(31).map(_ >> Integer.numberOfLeadingZeros(n))
+      else {
+        def loop: UIO[Int] =
+          next(31).flatMap { i =>
+            val value = i % n
+            if (i - value + (n - 1) < 0) loop
+            else UIO.succeed(value)
+          }
+        loop
+      }
 
-    private def nextRandom[T](shift: Data => (T, Data)) =
+    val nextLong: UIO[Long] =
```

Comment:
Did you push this up into `Random.Service`?

Reply:
No. Shall I do that?

Reply:
Yes, please! 🙏

Reply:
Done!